### ISSUE TITLE
Add `x-elastic-product-origin` header to serverless testing in `8.x` and `9.0`

### ISSUE THREAD

**Issue by robbavey** [elastic/logstash#16736]
Add Header to uses of the Kibana API endpoint `api/logstash/pipeline/*`

**Comment by robbavey** [comment]
Moved to https://github.com/elastic/logstash/issues/16736

**Comment by donoghuc** [comment]
I may be missing something here... I put up a draft of how I interpreted this task https://github.com/elastic/logstash/pull/16766 but i realize i may be way off.

**Comment by kaisecheng** [comment]
The header should be capital letter `X`, `X-elastic-product-origin`

**Comment by donoghuc** [comment]
Thanks for looking. All the tickets related to this work on the LS side and corresponding work so far have had the all lower case version. I did a search in the org and see both https://github.com/search?q=org%3Aelastic+x-elastic-product-origin&type=code&p=3 Header fields are case insensitive https://www.rfc-editor.org/rfc/rfc9110.html#fields but I agree consistency is probably nice. Is there a technical reason we would want to try to update everything with the capitalized first letter? 

The main question I have on this issue is that all the other tasks have been pretty clear in terms of ensuring a header is included for requests to an internal product. This has mostly consisted of finding http clients in the source code and ensuring that if it s communicating with an internal end point that it is including a header. This ticket references 
> the Kibana API endpoint api/logstash/pipeline/*
Which the only place in the LS source code i see using that endpoint is a shell script that appearst to be used in testing. I raised https://github.com/elastic/logstash/pull/16766 showing a change that would update those tests to send along the headers. 

So my two explicit questions now are:
1. Do we need to replace `x-elastic-product-origin` with `X-elastic-product-origin` everywhere? 
2. Is the intent of this ticket to send along the header in this script used in testing https://github.com/elastic/logstash/pull/16766 or is there something else i'm missing?
